packages/assets-controllers/src/AccountTrackerController.ts

```diff
@@ -282,69 +282,113 @@ export class AccountTrackerController extends StaticIntervalPollingController<Ac
+      (networkConfiguration) =>
+        networkConfiguration.rpcEndpoints.map(
+          (rpcEndpoint) => rpcEndpoint.networkClientId,
+        ),
+    );
+  }
+
   /**
    * Refreshes the balances of the accounts using the networkClientId
    *
    * @param input - The input for the poll.
-   * @param input.networkClientId - The network client ID used to get balances.
+   * @param input.networkClientIds - The network client IDs used to get balances.
    */
   async _executePoll({
-    networkClientId,
+    networkClientIds,
   }: AccountTrackerPollingInput): Promise<void> {
     // TODO: Either fix this lint violation or explain why it's necessary to ignore.
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    this.refresh(networkClientId);
+    this.refresh(networkClientIds);
   }
 
   /**
    * Refreshes the balances of the accounts depending on the multi-account setting.
    * If multi-account is disabled, only updates the selected account balance.
    * If multi-account is enabled, updates balances for all accounts.
    *
-   * @param networkClientId - Optional networkClientId to fetch a network client with
+   * @param networkClientIds - Optional network client IDs to fetch a network client with
    */
-  async refresh(networkClientId?: NetworkClientId) {
+  async refresh(networkClientIds: NetworkClientId[]) {
     const selectedAccount = this.messagingSystem.call(
       'AccountsController:getSelectedAccount',
     );
     const releaseLock = await this.#refreshMutex.acquire();
     try {
-      const { chainId, ethQuery } =
-        this.#getCorrectNetworkClient(networkClientId);
-      this.syncAccounts(chainId);
-      const { accountsByChainId } = this.state;
-      const { isMultiAccountBalancesEnabled } = this.messagingSystem.call(
-        'PreferencesController:getState',
-      );
-
-      const accountsToUpdate = isMultiAccountBalancesEnabled
-        ? Object.keys(accountsByChainId[chainId])
-        : [toChecksumHexAddress(selectedAccount.address)];
-
-      const accountsForChain = { ...accountsByChainId[chainId] };
-      for (const address of accountsToUpdate) {
-        const balance = await this.#getBalanceFromChain(address, ethQuery);
-        if (balance) {
-          accountsForChain[address] = {
-            balance,
-          };
-        }
-        if (this.#includeStakedAssets) {
-          const stakedBalance = await this.#getStakedBalanceForChain(
-            address,
-            networkClientId,
+      // Create an array of promises for each networkClientId
+      const updatePromises = networkClientIds.map(async (networkClientId) => {
+        const { chainId, ethQuery } =
+          this.#getCorrectNetworkClient(networkClientId);
+        this.syncAccounts(chainId);
+        const { accountsByChainId } = this.state;
+        const { isMultiAccountBalancesEnabled } = this.messagingSystem.call(
+          'PreferencesController:getState',
+        );
```

Comment:
nit - wondering if some of these stuff could be moved out of the loop?

We can do a refactor later.